bobcat/bb-junit/src/main/java/com/cognifide/qa/bb/junit/TestRunner.java

```diff
@@ -146,6 +158,39 @@ protected final void validateZeroArgConstructor(List<Throwable> errors) {
     // empty
   }
 
+  private void runMethod(FrameworkMethod method) throws Throwable {
+    int runs = 0;
+    int retryCount = retrieveRetry(method);
+    boolean testFailed = true;
+    while (testFailed) {
+      try {
+        methodBlock(method).evaluate();
+        testFailed = false;
+      } catch (AssumptionViolatedException e) {
+        throw e;
+      } catch (Throwable e) {
+        if (runs >= retryCount) {
+          throw e;
+        }
+        testEventCollector.removeLastEntry();
+      }
+      runs++;
+    }
+    if(runs>1){
+      testEventCollector.info("Test passed in: " + runs + " attempt");
+    }
+  }
+
+  private int retrieveRetry(FrameworkMethod method) {
+    Retry retry = method.getAnnotation(Retry.class);
+    return retry != null ? calculateRetryCount(retry.reruns()) : 0;
+  }
+
+  private int calculateRetryCount(int reruns) {
+    return reruns > 0 ? reruns
+        : Integer.parseInt(properties.getProperty(ConfigKeys.JUNIT_RERUNS, "0"));
```

Comment:
the default here is zero but for `@Retry` is -1. Why such discrepancy? IMO -1 is somehow misleading and unclear for me, I cannot see the difference between 0 and -1 in the aspect of reruning tests